pyiode/pyiode_model.pyx

```diff
@@ -6,45 +6,246 @@
+                    endo_exo_list=None,
+                    eps: float = 0.0001, 
+                    relax: float = 1.0, 
+                    maxit: int = 100, 
+                    init_values: int = KV_INIT_TM1, 
+                    sort_algo: int = SORT_BOTH, 
+                    nb_passes: int = 5, 
+                    debug: int = 0, 
+                    newton_eps: float = 1e-6, 
+                    newton_maxit: int = 50, 
+                    newton_debug: int = 0):
     '''
     Simulate the model defined by eqs_list on the period [sample_from, sample_to].     
     '''
-    if type(eqs_list) == list:
+    if isinstance(eqs_list, list):
         eqs_list = ','.join(eqs_list)
     elif eqs_list is None:
         eqs_list = ""
 
-    if type(endo_exo_list) == list:
+    if isinstance(endo_exo_list, list):
         endo_exo_list = ','.join(endo_exo_list)
     elif endo_exo_list is None:
         endo_exo_list = ""
         
+    model_simulate_save_parms(eps, relax, maxit, init_values, sort_algo, nb_passes)
+    
     if IodeModelSimulate(cstr(sample_from), cstr(sample_to), 
                          cstr(eqs_list), cstr(endo_exo_list),
                          eps, relax, maxit, init_values, sort_algo, nb_passes, debug, 
                          newton_eps, newton_maxit, newton_debug):
         raise RuntimeError(f"Simulation failed")
-   
+        
+    
+  
+def model_calc_scc(nb_passes: int = 1, 
+                   pre_listname: str = "_PRE", 
+                   inter_listname: str = "_INTER", 
+                   post_listname: str = "_POST", 
+                   eqs_list = None):
+    '''
+    Decompose a model into its Strong Connex Components: pre-recursive, interdependent
+    and post-recursive list of equations.
+    
+    These 3 components are saved in 3 lists whose names are given as parameters pre_listnames,
+    inter_listname and post_listname. 
+    
+    Optionally, the interdependent list of equations may be reordered by applying an heuristic
+    algorithm which tries to improve the position of the equations in the interdep list
+    to accelerate the Gauss-Seidel algorithm.
     
+    Parameters
+    ----------
+    nb_passes: int
+        number of iterations of the reordering algo (aka triangulation)
+    pre_listname: str
+        name of the iode list where the prolog block of equations must be stored
+    inter_listname: str 
+        name of the iode list where the interdependent block of equations must be stored
+    post_listname: str 
+        name of the iode list where the epilog block of equations must be stored
+    eqs_list
+        list of equations defining the model or None for all equations in the current WS of equations
+        
+    Returns
+    -------    
+    No return value.  On error, raises an exception .
+    
+    Examples
+    --------
+    >>> iode.model_calc_scc(nb_passes=1)        # doctest: +SKIP
     
+    '''
+    if isinstance(eqs_list, list):
```

Comment:
if isinstance(eqs_list, Iterable)

Reply:
Does not work. Same reason as in pyiode_estim.